Sources/Http/SignHashClient.swift

```diff
@@ -17,33 +17,42 @@ import Foundation
 
 final actor SignHashClient {
     
-    static func makeRequest(for request: SignHashRequest, accessToken: String, oauth2BaseUrl: String) async throws -> SignHashResponse {
+    static func makeRequest(for request: SignHashRequest, accessToken: String, oauth2BaseUrl: String) async throws -> Result<SignHashResponse, ClientError> {
         
         let endpoint = "/csc/v2/signatures/signHash"
         let baseUrl = oauth2BaseUrl + endpoint
 
         guard let url = URL(string: baseUrl) else {
-            throw ClientError.invalidRequestURL
+            return .failure(ClientError.invalidRequestURL)
         }
 
         var urlRequest = URLRequest(url: url)
```

Comment:
This is used throught the code base. Maybe Have an extension or function that creates this URL request?